tests/gis_tests/rasterapp/test_rasterfield.py

```diff
@@ -51,6 +51,23 @@ def test_access_band_data_directly_from_queryset(self):
         qs = RasterModel.objects.all()
         qs[0].rast.bands[0].data()
 
+    def test_deserialize_with_pixeltype_flags(self):
+        no_data = 3
+        rast = GDALRaster({
+            "srid": 4326,
+            "origin": [0, 0],
+            "scale": [-1, 1],
+            "skew": [0, 0],
+            "width": 1,
+            "height": 1,
+            "nr_of_bands": 1,
+            "bands": [{"data": [no_data], 'nodata_value': no_data}],
+        })
+        r = RasterModel.objects.create(rast=rast)
+        RasterModel.objects.filter(id=r.id).update(rast=Func(F('rast'), 1, function='ST_SetBandIsNoData'))
```

Comment:
Can create a regression test without using PostGIS-specific function?

Reply:
It may not be possible without a database roundtrip. I'm just asking.

Reply:
TBH, I don't know. You are more experienced than me.
I have rebased the PR with master.
I can try to find a way(maybe).